boards/arm/xenvm/xenvm.dts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2020 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This file was created by running
+ * # LIBXL_DEBUG_DUMP_DTB=domu-libxl.dtb xl create zephyr.conf
+ * decompilling resulting domu-libxl.dtb and then munally aligning it
+ * with zephyr requirements.
+ */
+
+/dts-v1/;
+
+#include <arm/armv8-a.dtsi>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+
+/ {
+	model = "XENVM-4.15";
+	compatible = "xen,xenvm-4.15\0xen,xenvm";
+	interrupt-parent = <&gic>;
+	#address-cells = <0x02>;
+	#size-cells = <0x02>;
+
+	chosen {
+		zephyr,console = &sbsa;
+		zephyr,shell-uart = &sbsa;
+	};
+
+	cpus {
+		#address-cells = <0x01>;
+		#size-cells = <0x00>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,armv8";
+			enable-method = "psci";
+			reg = <0x00>;
+		};
+	};
+
+	psci {
+		compatible = "arm,psci-1.0\0arm,psci-0.2\0arm,psci";
+		method = "hvc";
+		cpu_off = <0x01>;
+		cpu_on = <0x02>;
+	};
+
+	memory@40000000 {
+		device_type = "memory";
+		reg = <0x00 0x40000000 0x00 0x10000000>;
+	};
+
+	gic: interrupt-controller@3001000 {
+		compatible = "arm,cortex-a15-gic", "arm,cortex-a9-gic", "arm,gic";
+		label = "GIC";
+		#interrupt-cells = <0x04>;
+		#address-cells = <0x00>;
+		interrupt-controller;
+		reg = <0x00 0x3001000 0x00 0x1000 0x00 0x3002000 0x00 0x2000>;
+	};
+
+	timer {
+		compatible = "arm,armv8-timer", "arm,arm-timer";
+		interrupts = <GIC_PPI 0x0d IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY
+			      GIC_PPI 0x0e IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY
+		              GIC_PPI 0x0b IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>;
+		interrupt-parent = <&gic>;
+		label = "arch_timer";
+	};
+
+	hypervisor: hypervisor@38000000 {
+		compatible = "xen,xen-4.15\0xen,xen";
+		reg = <0x00 0x38000000 0x00 0x1000000>;
+		interrupts = <GIC_PPI 0x0f IRQ_TYPE_EDGE IRQ_DEFAULT_PRIORITY>;
+		interrupt-parent = <&gic>;
+	};
+
+	sbsa: sbsa-pl011@22000000 {
+		compatible = "arm,sbsa-uart", "arm,pl011";
```

Comment:
An SBSA driver would be able to drive a PL011 UART. But a PL011 driver may not be able to drive an SBSA UART because not all SBSA UART are PL011. For instance, the Xen emulation is a cut-down version.

So I am not sure this is correct to add the property ``arm,pl011``. Note that AFAICT, the toolstack doesn't expose ``arm,pl011``.

Reply:
Yes, you are right.
I think right solution is to add new compatible value to existing pl011 driver in Zephyr.